bus-mapping/src/evm/opcodes/tload.rs

```diff
@@ -0,0 +1,137 @@
+use super::Opcode;
+use crate::{
+    circuit_input_builder::{CircuitInputStateRef, ExecStep},
+    operation::{CallContextField, TransientStorageOp, RW},
+    Error,
+};
+use eth_types::{GethExecStep, ToWord, Word};
+
+/// Placeholder structure used to implement [`Opcode`] trait over it
+/// corresponding to the [`OpcodeId::TLOAD`](crate::evm::OpcodeId::TLOAD)
+/// `OpcodeId`.
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Tload;
+
+impl Opcode for Tload {
+    fn gen_associated_ops(
+        state: &mut CircuitInputStateRef,
+        geth_steps: &[GethExecStep],
+    ) -> Result<Vec<ExecStep>, Error> {
+        let geth_step = &geth_steps[0];
+        let mut exec_step = state.new_step(geth_step)?;
+
+        let call_id = state.call()?.call_id;
+        let contract_addr = state.call()?.address;
+
+        state.call_context_read(
+            &mut exec_step,
+            call_id,
+            CallContextField::TxId,
+            Word::from(state.tx_ctx.id()),
+        )?;
+
+        state.call_context_read(
+            &mut exec_step,
+            call_id,
+            CallContextField::CalleeAddress,
+            contract_addr.to_word(),
+        )?;
+
+        // First stack read
+        let key = geth_step.stack.last()?;
+        let stack_position = geth_step.stack.last_filled();
+
+        // Manage first stack read at latest stack position
+        state.stack_read(&mut exec_step, stack_position, key)?;
+
+        // Transient Storage read
+        let (_, value) = state.sdb.get_transient_storage(&contract_addr, &key);
+        let value = *value;
```

Comment:
It looks like we can do this
```suggestion
        let (_, &value) = state.sdb.get_transient_storage(&contract_addr, &key);
```